AC3.2Unit2FinalAssessment/AC3.2Unit2FinalAssessment/ColorTableViewController.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  ColorTableViewController.swift
+//  AC3.2Unit2FinalAssessment
+//
+//  Created by Amber Spadafora on 10/6/16.
+//  Copyright © 2016 C4Q. All rights reserved.
+//
+
+import UIKit
+
+class ColorTableViewController: UITableViewController {
+    
+    var numOfcells = crayolaColors.count
+    internal let rawColorData: [[String : Any]] = crayolaColors
+    internal var colorData: [Crayon]?
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        var colorContainer: [Crayon] = []
+        for rawColor in rawColorData {
+            colorContainer.append(Crayon(fromDict: rawColor)!)
+        }
+        colorData = colorContainer
+
+    }
+
+
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return numOfcells
+    }
+
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let cell = tableView.dequeueReusableCell(withIdentifier: "colorCell", for: indexPath)
+        
+        if let colorCell: ColorTableViewCell = cell as? ColorTableViewCell {
```

Comment:
Easier/cleaner to assign entire Crayon to one variable.

let crayon = colorData[indexPath.row]

from there you can grab _crayon.name_, _crayon.blue_ etc ...